src/main/java/com/rackspace/jenkins_nodepool/NodeRequest.java

```diff
@@ -118,6 +118,10 @@ public RequestState getState() {
     public List<NodePoolNode> getAllocatedNodes() throws Exception {
         // Example fulfilled request
         // {"nodes": ["0000000000"], "node_types": ["debian"], "state": "fulfilled", "declined_by": [], "state_time": 1520849225.4513698, "reuse": false, "requestor": "NodePool:min-ready"}
+
+        // Refresh our view of the data
+        updateFromZK();
```

Comment:
I've been somewhat reluctant to bring in NodePool as another (somewhat heavy) dependency - my approach was to minimize this since the Watcher class is such a simple class. Ideally, we could separate the ZK logic of add/update/delete path/data into a separate "handler" class which could be abstracted out (and perhaps replaced with a different implementation) and leave NodePool to the basic stuff such as provision/cleanup/node management.  Seems logical and sexy in theory, but perhaps not necessary in practice.